hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -7,21 +7,212 @@
+      desc: '''
+            Sanity + enable all interrupts and verify correctness of assertion and de-assertion of
+            each pin. Replace polling with waits on interrrupts.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_ctrl_arb
+      desc: '''
+            Continuously read data from host interface while randomly programming or reading from
+            the flash controller. Ensure the expected bandwidths are seen on both interfaces.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_read_collide
+      desc: '''
+            Read LSB bus word via host and program or erase the MSB bus word at the same time. Read
+            the MSB bus word within the same flash data word - it should read back the updated data,
+            proving that the internal buffer was cleared.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: en_mp_regions
+      desc: '''
+            Protect randomized and legal ranges of data and info partitions and ensure that the
+            corresponding read / program / erase enable bits are working. TBD - future support for
+            sub-page protection may be added - that will be tracked as a part of this test as well.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: program_erase_verify
+      desc: '''
+            TBD - alternative mechanisms of program and erase verification may be added in future -
+            this is a placeholder to verify that feature.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: redundant_pages
+      desc: '''
+            TBD - support for adding redundant pages may be added in future for yield recovery. THis
+            is a placeholder to verify that feature.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: lc_ctrl_if
+      desc: '''
+            TBD - verify the interface between flash controller and the life cycle controller.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: key_mgr_if
+      desc: '''
+            TBD - verify the seed output to the key manager.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: scramble
+      desc: '''
+            Enable scrambling, along with randomized scramble keys. Program a fresh chunk of
+            memory and read back (both, via controller and host) to ensure data integrity. On
+            program, verify via backdoor scrambling was done one the raw data correctly. When
+            reading via host, read the same memory via host multiple times back-to-back and ensure
+            the timing is correct (subsequent reads should be faster). When scrambling is not
+            enabled, ensure that the raw data is written and read back. Also randomly enable ECC.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: ecc
```

Comment:
this is something i did not make clear in my docs yet. At the moment, ecc and scrambling are enabled and disabled together.  So maybe this test and the above should be the same?

Reply:
Ohh, I see. I think we can leave it as two separate test points. The purpose of this doc it to only track the different features being tested. These test points can have identical `tests` key (which would mean the same set of tests verify both features / test points).